src/scripts/main.js

```diff
@@ -1,3 +1,30 @@
 'use strict';
 
-// write your code here
+const logo = document.querySelector('.logo');
+
+if (logo) {
+  const messageDiv = document.createElement('div');
+
+  messageDiv.classList.add('message');
+
+  function handleLogoClick() {
+    setTimeout(() =>{
+    const promise1 = Promise.resolve('Promise was resolved!');
+
+      promise1.then((result) => {
+        messageDiv.innerText = result;
+        document.body.appendChild(messageDiv);
+      });
+    }, 0)
+
+    setTimeout(() => {
+      const promise2 = Promise.reject(new Error('Promise was rejected!'));
+      promise2.catch((error) => {
+        messageDiv.classList.add('error-message');
+        messageDiv.innerText = error.message;
+      });
+    }, 3000);
+  }
```

Comment:
The `handleLogoClick` function might be doing too much. It creates a promise, sets its success and error handlers, and also controls the timing of these operations. Consider breaking this function down into smaller functions, each with a single responsibility. This will make your code easier to read and maintain.